console/v1/types_console_log_links.go

```diff
@@ -0,0 +1,38 @@
+package v1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ConsoleExternalLogLink is an extension for customizing OpenShift web console pod log links.
+type ConsoleExternalLogLink struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              ConsoleExternalLogLinkSpec `json:"spec"`
+}
+
+// ConsoleExternalLogLinkSpec is the desired console pod log link configuration.
+type ConsoleExternalLogLinkSpec struct {
+	// text is the display text for the link
+	Text string `json:"text"`
+	// hrefTemplate is an absolute secure URL (must use https) for the log link including
+	// variables to be replaced. Supported variables, which must exactly match, include:
+	// - resourceName - name of the pod which containes the logs
+	// - resourceUID - UID of the pod which contains the logs
+	//               - e.g. `11111111-2222-3333-4444-555555555555`
+	// - containerName - name of the pod's container that contains the logs
+	// - resourceNamespace - namespace of the pod that contains the logs
+	// - podLabels - JSON representation of labels matching the pod with the logs
+	//             - e.g. `{"key1":"value1","key2":"value2"}`
+	//
+	// e.g., https://example.com/logs?resourceName=\${resourceName}&containerName=\${containerName}&resourceNamespace=\${resourceNamespace}&podLabels=\${podLabels}
+	HrefTemplate string `json:"hrefTemplate"`
+	// namespaceFilter is a regular expression used to restrict a log link to a
+	// matching namespace (e.g., ^openshift-). If not specified, links will be
+	// displayed for all the namespaces.
+	// + optional
+	NamespaceFilter string `json:"namespaceFilter,omitempty"`
```

Comment:
Do i always get the k8s log link, and then if the namespacefilter matches, i get an additional (external) log link?

What happens if i have multiple ConsoleExternalLogLink objects, for an overlapping set of namespacefilters?

Reply:
> Do i always get the k8s log link, and then if the namespacefilter matches, i get an additional (external) log link?

I believe that you only get the link that  matches the `namespaceFilter`

> What happens if i have multiple ConsoleExternalLogLink objects, for an overlapping set of namespacefilters?

Not really sure with this one, @rhamilto cau you assist on this one ?

Reply:
> What happens if i have multiple ConsoleExternalLogLink objects, for an overlapping set of namespacefilters?

You get multiple links.  Note the filter is optional.  You'd get as many links as you have instances that aren't filtered in addition to any instances that match the filter.

Reply:
@bparees ˆˆ

Reply:
ok.  i'm a little surprised we remove the k8s log link when a custom log link exists, seems like that could still be useful? 

But i assume you guys have been through this w/ users/customers/use-cases?

Reply:
@bparees This link is on the pod log page. So you're already viewing the pod log, but have a link out to Kibana or another external logging solution. Bascially it's parity with what we have in 3.x:

https://docs.okd.io/3.11/install_config/web_console_customization.html#extension-option-for-external-logging-solutions

@jhadvig I think we should make it more clear where the log link will appear.

Reply:
got it.  sgtm.